raiden/network/proxies/user_deposit.py

```diff
@@ -101,19 +100,70 @@ def deposit(
         error_prefix = "Call to deposit will fail"
 
         with self.deposit_lock:
-            amount_to_deposit, log_details = self._deposit_preconditions(
-                total_deposit=total_deposit,
-                beneficiary=beneficiary,
-                token=token,
-                block_identifier=block_identifier,
-            )
+            # check preconditions
+            try:
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=given_block_identifier
+                )
+                current_balance = token.balance_of(
+                    address=self.node_address, block_identifier=given_block_identifier
+                )
+            except (BadFunctionCallOutput, ValueError):
+                # If 'given_block_identifier' has been pruned, we cannot perform the
+                # precondition checks but must still set the amount_to_deposit to a
+                # reasonable value.
```

Comment:
This is in-correct, if `given_block_identifier` is pruned, we skip all precondition checks altogether.

Reply:
I am skipping all precondition checks. `amount_to_deposit` is needed for the call to `token.approve()`, i. e. for the transaction.